### PR TITLE
Internal functions to extract namespaces from document

### PR DESCRIPTION
**Pull request by yogat3ch**
Add `doc_packages` and `namespace_exports` that extract all package names from a character vector and retrieves the exports for those packages respectively.

### CODE REVIEW

File: R/pretty_utils.R [hunk 1]
```diff
@@ -20,6 +20,98 @@ pretty_parse <- function(txt){
...
+
+#' @title namespace_exports
+#' @keywords Internal
+#' @description given a named character vector of packages outputs a list of all package exports
+#' @param ns \code{(character)} vector of package (namespace) names
+#' @return \code{(list)} of character vectors containing all of a package's exported functions
+#' @importFrom utils packageVersion install.packages select.list
+
+namespace_exports <- function(ns) {
+  known <- ip[,"Package"][ip[,"Package"] %in% ns]
```
Comment (yonicd) [inline R/pretty_utils.R:92]: where is IP coming from? I cant locate it as being defined anywhere else.

File: R/pretty_utils.R [hunk 2]
```diff
@@ -20,6 +20,98 @@ pretty_parse <- function(txt){
   
 }
 
+#' @importFrom stringi stri_opts_brkiter
+opts <- function (x) 
+{
+  if (identical(x, "")) {
+    stringi:::stri_opts_brkiter(type = "character")
```
Comment (yonicd) [inline R/pretty_utils.R:27]: `:::` throws a note and wont be accepted by CRAN. please replace with `stri_opts_brkiter <- utils::getFromNamespace('stri_opts_brkiter','stringi')`

File: R/pretty_rmd.R [hunk 3]
```diff
@@ -30,6 +32,7 @@ pretty_rmd <- function(input,
                        open_output = TRUE, 
                        create_library = TRUE,
                        chunks = NULL,
+                       lookup_order = c("searchpath", "document", "library", "sos"),
```
Comment (yonicd) [inline R/pretty_rmd.R:35]: This isnt being used in the function itself. what is the purpose of it?

**Review by yonicd — CHANGES_REQUESTED** [review]
Apologies for disappearing on this PR, new job took precedence over all open source.

 I just checked it out. 

It looks good, just so i understand, these additions are not exported and are not used in the package beyond being defined as far as i can see.

Three things are not right that I commented in the review.

File: R/pretty_rmd.R (hunk 3 above)
Comment (yogat3ch) [reply R/pretty_rmd.R:35]: Hey @yonicd ,
This was a feature that I had intended to build out but never did :-/ 
It was meant to allow the user to specify the priority of where packages are searched for (and omit some if they wish)

Comment (yonicd) [reply R/pretty_rmd.R]: Ah ok. I thought that that was the intention, but i did see it implemented in the function after :)

File: R/pretty_utils.R [hunk 4]
```diff
@@ -20,6 +20,81 @@ pretty_parse <- function(txt){
...
+  }
+}
+
+#' @title doc_packages
+#' @keywords Internal
+#' @description A function that finds all packages in a document based on typical methods of loading namespaces in R
+#' @param txt \code{(character)} vector of document text
+#' @details See \href{https://github.com/yonicd/sinew/issues/61}{#61} for a discussion
+#' @return \code{(character)} vector of all packages loaded in the doc
+
```
[the remark this answers is not in the document]
Comment (yonicd) [reply R/pretty_utils.R:40]: I am extra verbose in this so it is clearer in the documentation which functions are being imported in the function. 

This is also the motivation for make_import() and make_oxygen().

**Comment by yonicd** [comment]
After thinking about the functions in the PR I am now leaning on not including them. Sorry for the extra work you did, I am really appreciative for it. Some reasoning:

- `namespace_export` since it does the same thing as `make_force_packages` and was PR'ed the other day.
- `opt` isnt used in the rest of the package and doesnt serve a purpose that i can see.

With regards to doc_packages, why is it internal?

**Pull request closed without merging**